frontend/src/libraries/explorer-wamp/accounts.ts

```diff
@@ -74,15 +74,20 @@ export default class AccountsApi extends ExplorerApi {
     }
   }
 
-  async getAccounts(limit: number = 15): Promise<AccountBasicInfo[]> {
+  async getAccounts(
+    limit: number = 15,
+    endTimestamp: number = -1
+  ): Promise<AccountBasicInfo[]> {
     try {
       return await this.call("select", [
         `SELECT account_id as id, timestamp, transaction_hash as address 
-        FROM accounts 
+        FROM accounts
+        WHERE timestamp < :endTimestamp
         ORDER BY timestamp DESC
         Limit :limit`,
         {
-          limit
+          limit,
+          endTimestamp: endTimestamp === -1 ? "MAX(timestamp)" : endTimestamp
```

Comment:
Well, you, essentially, create a more complex query by requesting MAX value and filtering for it later for nothing. Also, you exclude the last/MAX value. Thus, if there is no `endTimestamp`, there is no need to have the WHERE clause.